scripts/pipelines/log-buildversions.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+# This script creates a file documenting the commits
+# used to during the most recent build. The file
+# includes both the commits used by AUTOREV packages
+# and the commits used for the git submodules.
+
+SCRIPT_ROOT=$(realpath $(dirname $BASH_SOURCE))
+
+function usage() {
+	cat <<EOF
+$(basename $0) [-h] [-o FILEPATH]
+
+Get information on the specific commit versions for NILRT components after a build.
+
+# Options
+-h
+	Print this message and exit.
+-o
+	The filepath to output the commit version information to.
+-a
+	Include all SRCREVs and not just AUTOREVs.
+EOF
+}
+
+logfile_path=
+buildhistory_opts=
+while getopts ":ho:a" opt; do
+	case $opt in
+		h )
+			usage >&1
+			exit 0
+			;;
+		o )
+			if [ ${OPTARG} = "/*" ]; then
+				logfile_path="$OPTARG"
+			else
+				logfile_path="$SCRIPT_ROOT/$OPTARG"
+			fi
+			;;
+		a )
+			buildhistory_opts="-a"
+			;;
+		\: )
+            echo "ERROR: Missing argument for option -${OPTARG}" >&2
+			exit 1
+            ;;
+		\? )
+			echo "ERROR: Invalid option -${OPTARG}." >&2
+			exit 1
+			;;
+	esac
+done
+shift $(($OPTIND - 1))
+
+# Gets nilrt and submodule repo hashes.
+function get_repo_hashes() {
+	local nilrt_rev=$(git rev-parse HEAD)
+	local submodule_status=$(git submodule status)
+	cat <<EOF
+# git repo hashes for nilrt and submodules.
+# Checkout these commits to use the same versions as this build.
+nilrt commit id: $nilrt_rev
+
+# Submodules
+$submodule_status
+EOF
+}
+
+# Gets SRCREV values from the OE buildhistory.
+function get_rev_commits() {
+	local buildhistory_dir=buildhistory/
+	if [ -d buildhistory/ ]; then
+		local buildhistory_output="$(buildhistory-collect-srcrevs ${buildhistory_opts})"
+		cat <<EOF
+# Build revisions used for recipes. Copy into local.conf or a similar
+# global conf file to staple to the same revisions used for this build.
+$buildhistory_output
+EOF
+	else
+		>&2 echo "ERROR: No buildhistory directory at $buildhistory_dir. Make sure a build with buildhistory enabled occured."
```

Comment:
How do you enable buildhistory? And does that add a lot of extra time to the build?

Reply:
You enable buildhistory as seen in our `auto.conf` [here](https://github.com/ni/nilrt/blob/92fa4b60e79f72a96b06a7a31092be6dc50f528e/scripts/azdo/conf/auto.conf#L1). I'm not sure how much time it adds, but we already enable it in our AzDO builds (see [here](https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/rtos/includes/bitbakeRules.mak&version=GC66e787d6f0a2321189ac63243a62df45b7022c8c&line=97&lineEnd=102&lineStartColumn=1&lineEndColumn=76&lineStyle=plain&_a=contents)) so it isn't adding time to the build that isn't already added.